packages/server/graphql/mutations/emailPasswordReset.ts

```diff
@@ -53,12 +55,29 @@ const emailPasswordReset = {
           .count()
           .ge(Threshold.MAX_DAILY_PASSWORD_RESETS) as unknown) as boolean
       }).run()
-      if (failOnAccount || failOnTime || !user) return true
+      if (failOnAccount || failOnTime) {
+        return {error: {message: AuthenticationError.EXCEEDED_RESET_THRESHOLD}}
+      }
+      if (!user) return {error: {message: AuthenticationError.USER_NOT_FOUND}}
```

Comment:
let's do the SAML validation first:
- it's possible that the person hasn't created an account yet, but their domain is SAML, in which case we want them to see the SAML instructions
- it's possible the person used to log in with password or google but then their org upgraded to SAML, in which case they should go through the SAML flow

Additional tests:
- [ ] domain `foo.co` is a registered SAML domain
- [ ] when `foo@foo.co` who doesn't exist in our system clicks forget password, they get the SAML error
- [ ] when `bar@foo.co` who DOES exist in our system, they get the SAML error
- [ ] when `baz@foo.co` who has a google identity clicks, they get a SAML error